test_conformance/api/test_mem_object_properties_queries.cpp

```diff
@@ -0,0 +1,317 @@
+                context, test_case.properties.data(), flags,
+                local_data.size() * sizeof(cl_uint), local_data.data(), &error);
+        }
+
+        test_error(error, "clCreateBufferWithProperties failed.");
+    }
+    clGetMemObjectInfo(test_object, CL_MEM_PROPERTIES, 0, NULL, &set_size);
+    test_error(error,
+               "clGetMemObjectInfo failed asking for CL_MEM_PROPERTIES.");
+
+    // verify set_size 0 returned
+    if (test_case.properties.size() == 0 && set_size == 0)
+    {
+        return TEST_PASS;
+    }
+
+    cl_uint number_of_props = set_size / sizeof(cl_mem_properties);
+    object_properties_check.resize(number_of_props);
+    clGetMemObjectInfo(test_object, CL_MEM_PROPERTIES, set_size,
+                       object_properties_check.data(), NULL);
+    test_error(error,
+               "clGetMemObjectInfo failed asking for CL_MEM_PROPERTIES.");
+
+    // check list with 0 terminator is returned
+    if (object_properties_check.size() == 1 && object_properties_check[0] == 0
+        && test_case.properties.size() == 0)
+    {
+        return TEST_PASS;
+    }
+    if (object_properties_check.back() != 0)
+    {
+        log_error("ERROR: Incorrect last properties value - should be 0!\n");
+        return TEST_FAIL;
+    }
+    object_properties_check.pop_back();
+    test_case.properties.pop_back();
+    if (object_properties_check != test_case.properties)
+    {
+        for (cl_uint i = 0; i < test_case.properties.size(); i = i + 2)
+        {
+            cl_mem_properties set_property = test_case.properties[i];
+            cl_mem_properties set_property_value = test_case.properties[i + 1];
+            std::vector<cl_mem_properties>::iterator it =
+                std::find(object_properties_check.begin(),
+                          object_properties_check.end(), set_property);
+
+            if (it == object_properties_check.end())
+            {
+                log_error("ERROR: Property not found ... 0x%x\n", set_property);
+                return TEST_FAIL;
+            }
+            else
+            {
+                if (set_property_value != *std::next(it))
+                {
+                    log_error("ERROR: Incorrect preperty value expected %x, "
+                              "obtained %x\n",
+                              set_property_value, *std::next(it));
+                    return TEST_FAIL;
+                }
+            }
+        }
+        log_error(
+            "ERROR: ALL properties and values matched but order incorrect!\n");
+        return TEST_FAIL;
+    }
+
+    return error;
+}
+
+int run_test_query_properties(cl_context context, cl_command_queue queue,
+                              test_data test_case)
+{
+    int error = CL_SUCCESS;
+    log_info("\nTC description: %s\n", test_case.description.c_str());
+
+    clProgramWrapper program;
+    clKernelWrapper kernel;
+    clMemWrapper obj_src;
+    clMemWrapper obj_dst;
+    clEventWrapper event;
+    MTdata d;
```

Comment:
Another minor point, but would it might be helpful for future developers to give this variable a slightly more descriptive name? It could also be initialized when it is defined.